core/src/main/java/org/elasticsearch/ingest/core/IngestDocument.java

```diff
@@ -142,6 +154,16 @@ public IngestDocument(Map<String, Object> sourceAndMetadata, Map<String, String>
     }
 
     /**
+     * Checks whether the document contains a value for the provided templated path
+     * @param fieldPathTemplate the template for the path within the document in dot-notation
+     * @return true if the document contains a value for the field, false otherwise
```

Comment:
maybe this is outside the scope of this PR? as this `hasField` simply overrides the first to enable the `Template` argument